solr/core/src/java/org/apache/solr/search/JoinQParserPlugin.java

```diff
@@ -587,17 +587,27 @@ public DocSet getDocSetEnumerate() throws IOException {
             // if we don't have a bitset yet, or if the resulting set will be too large
             // use the filterCache to get a DocSet
             if (toTermsEnum.docFreq() >= minDocFreqTo || resultBits == null) {
-              // use filter cache
-              DocSet toTermSet = toSearcher.getDocSet(toDeState);
-              resultListDocs += toTermSet.size();
-              if (resultBits != null) {
-                toTermSet.addAllTo(new BitDocSet(resultBits));
-              } else {
-                if (toTermSet instanceof BitDocSet) {
-                  resultBits = ((BitDocSet)toTermSet).bits.clone();
+              try {
+                // use filter cache
+                DocSet toTermSet = toSearcher.getDocSet(toDeState);
```

Comment:
Shouldn't only this line be wrapped in try-catch for IllegalStateException?

Reply:
toTermSet is used in whole block, so it would introduce null checks